### PR TITLE
clean regex

### PR DESCRIPTION
**Pull request by caub**
use strict not needed either

### CODE REVIEW

**Comment by caub** [comment]
@killercup Thanks for that module, can I ask you why 'id' is not simply 'id' instead of 'id ASC NULLS LAST'. Why not omitting ASC and NULL LAST default values?

Also is it really injection-safe?

**Comment by killercup** [comment]
Thanks, but I'd rather keep the code very explicit. :) That's also the reason why it generates the full form for a simple `id`.

> Also is it really injection-safe?

I think so? The regex is very specific in allowing alphanumerical characters and dots only (aside from the plus/minus pre/suffix), so it'd be interesting to see if you can break SQL based on that…

Anyway, I haven't tested it or anything, nor even thought about it, as this was just for a internal prototype and didn't go to production or anything.

**Comment by killercup** [comment]
Oh, you can have it generate invalid SQL, of course, using something like `+foo......bar`. But you can supply invalid column names anyway, so it's not really a hack.

**Comment by caub** [comment]
I plan to use this module, so I just checked the source and made those little changes (it's clearer and very common to use `?` in the regex), the `use strict` are not necessary and that's it. Would be glad if you explain which part annoys you

I think it would be better without the defaults `NULL LAST` and `ASC`, but I respect your respect your choice there

File: index.js [hunk 1]
```diff
@@ -21,17 +21,15 @@ var sortStrFormat = /^([+-]|)([\w.]+)(-|)$/;
...
   }
 
-  var field = parsed[2];
-  var order = parsed[1] === '-' ? 'DESC' : 'ASC';
-  var nulls = parsed[3] === '-' ? 'NULLS FIRST' : 'NULLS LAST';
+  var order = parsed[1] === '-' ? ' DESC' : ' ASC';
+  var nulls = parsed[3] === '-' ? ' NULLS FIRST' : ' NULLS LAST';
 
-  return field + ' ' + order + ' ' + nulls;
+  return parsed[2] + order + nulls;
```
Comment (killercup) [inline index.js]: This really just makes it harder to read (I'm also quite sure it doesn't matter for performance)

**Review by killercup — CHANGES_REQUESTED** [review]
Okay, the `?` regex is totally fine, and getting rid of `use strict` is probably fine in 2017 as well. Removing the `field` var just makes it harder to read, though.

**Comment by killercup** [comment]
Thanks for your interest and wanting to help out here, by the way :)

> I think it would be better without the defaults NULL LAST and ASC, but I respect your respect your choice there

Cool. It's just that adding code to make things less explicit is not a good trade-off to me :)

Btw, What database driver are you using? Maybe it'd be a good idea to not just have this emit strings but rather an array of orderings? It might also be a good idea to not just silently ignore fields that don't match the regex.

To be honest, I really have no use case for this anymore, and if you want to, I could absolutely transfer the repo as well as the npm package to you. This way you don't have to wait for me and this package gets some more love :)

**Comment by caub** [comment]
updated

I use [pg](https://github.com/brianc/node-postgres/) module, and a string as input is good, because it would come like that from an api call query-string `&sort=-updated`

**Review by killercup — APPROVED** [review]

**Comment by killercup** [comment]
Thanks & merged! Just let me know if you want to maintain this package :)

**Comment by caub** [comment]
ok no problem, as you wish

**Comment by killercup** [comment]
You now have contributor rights on github and npm 👍

**Pull request merged**